### ISSUE TITLE
kernel dies just after starting if matplotlib is installed, works fine if not.

### ISSUE THREAD

**Issue by pedroangelini** [ipython/ipykernel#798]
When starting a notebook, if I have the package matplotlib installed in my environment, the notebook becomes unresponsive while running the first cell. This happens independently of which cell I'm running, even a simple `print("Hi")` will stay in the "running" state and not output anything.
If I pip uninstall matplotlib, the notebooks start working again, but of course, for most of my data science work I need matplotlib.
I've started a fresh installation of python (3.10) and tested adding and removing the packages I have in a separate computer, and tested uninstalling a few packages at a time. 

I'm attaching my jupyter-troubleshoot output, and also the debug log of a failed session. 
[jupyter-troubleshoot.txt](https://github.com/ipython/ipykernel/files/7521288/jupyter-troubleshoot.txt)
[notebooklog.txt](https://github.com/ipython/ipykernel/files/7521289/notebooklog.txt)

It seems that this error happens when things go wrong, but it does not appear when I remove matplotlib and try again:

`[D 08:47:00.499 NotebookApp] 200 GET /api/contents/merge_save_transports.ipynb/checkpoints?_=1636642019748 (::1) 20.940000ms
ERROR:asyncio:Exception in callback <TaskStepMethWrapper object at 0x000002382B54BD00>()
handle: <Handle <TaskStepMethWrapper object at 0x000002382B54BD00>()>
Traceback (most recent call last):
  File "C:\Users\usangelipe\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
RuntimeError: Cannot enter into task <Task pending name='Task-1' coro=<HTTP1ServerConnection._server_request_loop() running at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\tornado\http1connection.py:840> cb=[IOLoop.add_future.<locals>.<lambda>() at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\tornado\ioloop.py:688]> while another task <Task pending name='Task-74' coro=<MappingKernelManager.start_kernel() running at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\notebook\services\kernels\kernelmanager.py:176> cb=[IOLoop.add_future.<locals>.<lambda>() at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\tornado\ioloop.py:688]> is being executed.
ERROR:asyncio:Exception in callback <TaskStepMethWrapper object at 0x000002382B54A0E0>()
handle: <Handle <TaskStepMethWrapper object at 0x000002382B54A0E0>()>
Traceback (most recent call last):
  File "C:\Users\usangelipe\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
RuntimeError: Cannot enter into task <Task pending name='Task-33' coro=<HTTP1ServerConnection._server_request_loop() running at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\tornado\http1connection.py:840> cb=[IOLoop.add_future.<locals>.<lambda>() at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\tornado\ioloop.py:688]> while another task <Task pending name='Task-74' coro=<MappingKernelManager.start_kernel() running at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\notebook\services\kernels\kernelmanager.py:176> cb=[IOLoop.add_future.<locals>.<lambda>() at C:\Users\usangelipe\AppData\Roaming\Python\Python310\site-packages\tornado\ioloop.py:688]> is being executed.
[D 08:47:00.545 NotebookApp] Instantiating kernel 'Python 3 (ipykernel)' with kernel provisioner: local-provisioner`

**Comment by blink1073** [comment]
Hi @pedroangelini, I think this is related to https://github.com/jupyter/notebook/pull/6221, and should be fixed once that is merged and a new version of `notebook` is released.

**Comment by dleen** [comment]
Should be fixed in `notebook==6.4.6` 👍
